### DISCUSSION TITLE
console.save_html -> Colors

### DISCUSSION THREAD

**Discussion by FedeClaudi** [Textualize/rich#946] [Q&A]
Hi Will, 

I'm implementing a method to export some stuff to .html in [refy](https://github.com/FedeClaudi/refy) but I'm struggling a bit with how to set up the theme to get the look I need. 
I defined a `TERMINAL_THEME` class like the one from [rich](https://github.com/willmcgugan/rich/blob/4508ed3a43/rich/terminal_theme.py) and I could set the background color of html page, but I can't seem to be able to change individual elements. 

In particular I would like to be able to set the color of the links, that are in black currently as you can see in the screenshot:
<img width="1568" alt="image" src="https://user-images.githubusercontent.com/17436313/105100205-dc2cec00-5aa4-11eb-952b-e6a9cc22cc13.png">

but I couldn't figure out how to do this via a custom `Theme` class. 

Any help would be much appreciated, 
thanks.
Fede

**Comment by FedeClaudi** [comment]
By the way I realize that I can change the color by styling the text first, printing it and then exporting to HTML, but I would like to know if in general there's an easier way to determine the final look of the HTML.

<img width="1598" alt="image" src="https://user-images.githubusercontent.com/17436313/105104408-164cbc80-5aaa-11eb-923b-d00d91625e83.png">

**Comment by willmcgugan** [comment]
Hi Federico,

The TerminalTheme defines the default foreground/background and the first 16 colors because those are generally set by the terminal, and Rich needs to know what they are when exporting html. For the other colors Rich will try to faithfully reproduce what you see on the terminal. I'm guessing your black text was the default color? Setting foreground in the theme should change that.

Other than the first 16 colors there's no way to modify the colors at the point of export. You would need to modify the colors you are printing. I guess you have found the [theme system](https://rich.readthedocs.io/en/latest/style.html#style-themes) which is the best way to do that?

One other way to modify the HTML output would be to set the `code_format` argument which is the HTML template. This is the [default](https://github.com/willmcgugan/rich/blob/master/rich/console.py#L61), but you could modify it with CSS rules. You could define a `code a` rule to tweak the links for instance.

Hope that helps.

**Comment by FedeClaudi** [comment]
Hi, 

Yes indeed the black text was the ones I had left as default, when I st them to a color they were saved as such in the HTML. 
I did use a custom theme but I didn't thin of trying to change the foreground, I was changing the other colors but I was confused as to what related to what. 

This helps a lot, thank you very much!
